siriuspy/siriuspy/devices/ids.py

```diff
@@ -834,6 +864,122 @@ def cmd_clear_error(self):
         pass
 
 
+class DELTA(_ID):
+    """DELTA Insertion Device."""
+
+    class DEVICES:
+        """Device names."""
+
+        DELTA52_10SB = 'SI-10SB:ID-DELTA52'
+        ALL = (DELTA52_10SB, )
+
+    PROPERTIES_DEFAULT = _ID.PROPERTIES_DEFAULT + (
+        'PolShift-Mon', 
+        'Pol-Mon',
+        'ChangePol-Cmd',
+        'GainShift-SP', 'GainShift-RB', 'GainShift-Mon',
+        'MaxVelo-SP', 'MaxVelo-RB',
+        'PolModeVelo-SP', 'PolModeVelo-RB',
+        'PolModeAcc-SP', 'PolModeAcc-RB',
+        'GainModeVelo-SP', 'GainModeVelo-RB',
+        'GainModeAcc-SP', 'GainModeAcc-RB',
+        )
+
+    def __init__(self, devname=None, props2init='all', auto_monitor_mon=True):
+        """."""
+        # check if device exists
+        if devname is None:
+            devname = self.DEVICES.DELTA52_10SB
+        if devname not in self.DEVICES.ALL:
+            raise NotImplementedError(devname)
+
+        # call base class constructor
+        super().__init__(
+            devname, props2init=props2init, auto_monitor_mon=auto_monitor_mon)
+
+    # --- pparameter ---
+
+    @property
+    def pparameter_speed_max(self):
+        """Return max pparameter speed readback [mm/s]."""
+        return self['MaxVelo-RB']
+
+    @property
+    def pparameter_speed_max_lims(self):
+        """Return max pparameter speed limits."""
+        ctrl = self.pv_ctrlvars('MaxVelo-SP')
+        lims = [ctrl['lower_ctrl_limit'], ctrl['upper_ctrl_limit']]
+        return lims
+
+    @property
+    def pparameter_speed(self):
+        """Return pparameter speed readback [mm/s]."""
+        return self['PolModeVelo-RB']
+
+    @property
+    def pparameter_min(self):
+        """Return ID pparameter lower control limit [mm]."""
+        ctrlvars = self.pv_ctrlvars('PolShift-Mon')
+        return ctrlvars['lower_ctrl_limit']
+
+    @property
+    def pparameter_max(self):
+        """Return ID pparameter upper control limit [mm]."""
+        ctrlvars = self.pv_ctrlvars('PolShift-Mon')
+        return ctrlvars['upper_ctrl_limit']
```

Comment:
why the min and max of pparameters were defined as two properties but the speed max lims as a single property returning a list with two values? It seems to break some pattern.

Reply:
deprecated